public/javascripts/rateRecipe.js

```diff
@@ -1,3 +1,90 @@
 document.addEventListener('DOMContentLoaded', () => {
-    
+  const getRecipeIdAndStarRating = (element) => {
+    const ids = element.id.split('.');
+    const recipeId = parseInt(ids[0].split(':')[1], 10);
+    const starRating = parseInt(ids[1].split(':')[1], 10);
+
+    return { recipeId, starRating };
+  };
+
+  const anyIsChecked = (recipeId, num) => {
+    for (let i = 1; i < num; i++) {
+      const star = document.getElementById(`recipe:${recipeId}.star:${i}`);
+      if (star.classList.contains('checked')) {
+        return true;
+      }
+    }
+    return false;
+  };
+
+  const fillStars = (recipeId, starRating) => {
+    for (let i = 1; i <= starRating; i++) {
```

Comment:
Is there a way to make this happen in just one for loop? I noticed that the first line of each loop is the same, so it seems like you could just make a conditional that determines whether to add or remove checked after that and DRY up this function a little.